src/SettingsRepositoryInterface.h

```diff
@@ -11,6 +11,7 @@ class cSettingsRepositoryInterface
 	};
 
 	cSettingsRepositoryInterface() = default;
+	cSettingsRepositoryInterface(cSettingsRepositoryInterface &&) = default;
```

Comment:
Base classes with virtual functions generally shouldn't be movable or copyable. That's a recipe for object slicing.